### PR TITLE
Fixed issue no 309

### PR DESCRIPTION
**Pull request by pawannec**
Fixed issue no 309

### CODE REVIEW

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java [hunk 1]
```diff
@@ -63,6 +63,9 @@ public static Map<String, Object> clearBaseProps(Map<String, Object> fullEntry,
 		if (attrNames == null || attrNames.isEmpty()) {
 			return fullEntry;
 		}
+		if(subscription.getSubscription().getNotification().getAttributeNames() == null || subscription.getSubscription().getNotification().getAttributeNames().isEmpty()) {
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java]: [Minor] Please take in List<String> and then compare null and isEmpty.

Comment (pawannec) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java]: Done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java [hunk 2]
```diff
@@ -63,6 +63,10 @@ public static Map<String, Object> clearBaseProps(Map<String, Object> fullEntry,
 		if (attrNames == null || attrNames.isEmpty()) {
 			return fullEntry;
 		}
+		List<String> notifucationAttrs = subscription.getSubscription().getNotification().getAttributeNames();
```
Comment (ScorpioBroker) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java]: [minor] typo "notification"

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java [hunk 3]
```diff
@@ -63,6 +63,10 @@ public static Map<String, Object> clearBaseProps(Map<String, Object> fullEntry,
 		if (attrNames == null || attrNames.isEmpty()) {
```
Comment (ScorpioBroker) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java]: remove attrsNames check here (62-65) as it checks for the watchedAttributes which we shouldn't

**Review by ScorpioBroker — CHANGES_REQUESTED** [review]

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java (hunk 2 above)
Comment (amitrnec) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java]: Done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java (hunk 3 above)
Comment (amitrnec) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/EntityTools.java]: Removed

**Review by ScorpioBroker — APPROVED** [review]
lgtm

**Pull request merged**